packages/blade-mcp/src/tools/createBladeCursorRules.ts

```diff
@@ -2,59 +2,83 @@ import { join } from 'path';
+  CURSOR_RULES_VERSION,
+  handleError,
+  sendAnalytics,
+} from '../utils.js';
 
-const createBladeCursorRulesDescription =
+const createBladeCursorRulesToolName = 'create_blade_cursor_rules';
+
+const createBladeCursorRulesToolDescription =
   'Creates the cursor rules for blade to help with code generation. Call this before get_blade_docs and while creating a new blade project (only when using cursor and when the frontend-blade-rules.mdc rule does not already exist).';
 
-const createBladeCursorRulesSchema = {
+const createBladeCursorRulesToolSchema = {
   currentProjectRootDirectory: z
     .string()
     .describe(
       "The working root directory of the consumer's project. Do not use root directory, do not use '.', only use absolute path to current directory",
     ),
 };
 
-const createBladeCursorRulesCallback: ToolCallback<typeof createBladeCursorRulesSchema> = ({
+const createBladeCursorRulesToolCallback: ToolCallback<typeof createBladeCursorRulesToolSchema> = ({
   currentProjectRootDirectory,
 }) => {
-  const ruleFileDir = join(currentProjectRootDirectory, '.cursor/rules');
-  const ruleFilePath = join(ruleFileDir, 'frontend-blade-rules.mdc');
-
-  if (existsSync(ruleFilePath)) {
-    if (hasOutDatedRules(ruleFilePath)) {
-      // removes the outdated rules file and continues execution to generate new rule file
-      unlinkSync(ruleFilePath);
-    } else {
-      return {
-        content: [{ type: 'text', text: 'Cursor rules already exist. Doing nothing' }],
-      };
+  try {
+    const ruleFileDir = join(currentProjectRootDirectory, '.cursor/rules');
+    const ruleFilePath = join(ruleFileDir, 'frontend-blade-rules.mdc');
+
+    if (existsSync(ruleFilePath)) {
+      if (hasOutDatedRules(ruleFilePath)) {
+        // removes the outdated rules file and continues execution to generate new rule file
+        unlinkSync(ruleFilePath);
+      } else {
+        return {
+          content: [{ type: 'text', text: 'Cursor rules already exist. Doing nothing' }],
+        };
+      }
     }
-  }
 
-  const ruleFileTemplateContent = readFileSync(BLADE_CURSOR_RULES_FILE_PATH, 'utf8').replace(
-    'rules_version: <!-- dynamic_version -->',
-    `rules_version: ${CURSOR_RULES_VERSION}`,
-  );
+    const ruleFileTemplateContent = readFileSync(BLADE_CURSOR_RULES_FILE_PATH, 'utf8').replace(
+      'rules_version: <!-- dynamic_version -->',
+      `rules_version: ${CURSOR_RULES_VERSION}`,
+    );
 
-  if (!existsSync(ruleFileDir)) {
-    mkdirSync(ruleFileDir, { recursive: true });
-  }
+    if (!existsSync(ruleFileDir)) {
+      mkdirSync(ruleFileDir, { recursive: true });
+    }
 
-  writeFileSync(ruleFilePath, ruleFileTemplateContent);
+    writeFileSync(ruleFilePath, ruleFileTemplateContent);
 
-  return {
-    content: [
-      {
-        type: 'text',
-        text: `Blade cursor rules created at: ${ruleFilePath}. Cursor Rules Version: ${CURSOR_RULES_VERSION}`,
+    sendAnalytics({
+      eventName: 'Blade MCP Tool Called',
```

Comment:
we can move this string to variable as well. Someone might mess up the spacing or spellings